Alerting: Add HMAC signature config to the webhook integration (#100960)
grafana/grafana: docs/sources/alerting/configure-notifications/manage-contact-points/integrations/webhook-notifier.md

| Authentication Header Credentials | Credentials for the `Authorization` Request header.                                                                     |
| Max Alerts                        | Maximum number of alerts to include in a notification. Any alerts exceeding this limit are ignored. `0` means no limit. |
| TLS                               | TLS configuration options, including CA certificate, client certificate, and client key.                                |
| HMAC Signature                    | HMAC signature configuration options.                                                                                   |

{{< admonition type="note" >}}

You can configure either HTTP Basic Authentication or the Authorization request header, but not both.

{{< /admonition >}}

> #### HMAC Signature

Reviewer: https://grafana.com/docs/writers-toolkit/write/style-guide/capitalization-punctuation/#sentence-case

```suggestion
#### HMAC signature
```

> You can secure your webhook notifications using HMAC signatures to verify the authenticity and integrity of the requests. When enabled, Grafana will sign the webhook payload with a shared secret using HMAC-SHA256.

Reviewer: Uses present tense:

```suggestion
You can secure your webhook notifications using HMAC signatures to verify the authenticity and integrity of the requests. When enabled, Grafana signs the webhook payload with a shared secret using HMAC-SHA256.
```

| Header           | The HTTP header where the signature will be set. Default is `X-Grafana-Alerting-Signature`.                                                                                                                                    |
| Timestamp Header | Optional header to include a timestamp in the signature calculation. When specified, Grafana will set a Unix timestamp in this header and include it in the HMAC calculation. This provides protection against replay attacks. |

When HMAC signing is configured, Grafana generates a signature using HMAC-SHA256 with your secret key. If a timestamp header is specified, a Unix timestamp is included in the signature calculation. The signature is calculated as:

```
HMAC(timestamp + ":" + body)
```

The timestamp is sent in the specified header. If no timestamp header is specified, the signature is calculated just from the request body. The signature is sent as a hex-encoded string in the specified signature header.

> ##### Validating a request

Reviewer: https://grafana.com/docs/writers-toolkit/review/lint-prose/rules/#grafanagerunds

```suggestion
##### Validate a request
```
